### PR TITLE
fix: gap on infinite scroll

### LINKED ISSUE #7642: Infinite scroll loading items without skeletons or borders, leaving gaps
### What happened?

Observe: 

https://github.com/kodadot/nft-gallery/assets/36627808/db5f0290-d8b2-4b35-8fa2-6102d6dc326a

### Please reproduce in steps

Explore items -> scroll down fast

### Expected Behavior

Should load in a organized way and gaps should not be present

### PR DESCRIPTION
**Pull request by preschian**
**Thank you for your contribution** to the [KodaDot - One Stop Shop for Polkadot NFTs](https://kodadot.xyz).

👇 __ Let's make a quick check before the contribution.

## PR Type

- [x] Bugfix
- [ ] Feature
- [ ] Refactoring

## Context

- [x] Closes #7642

image

#### Did your issue had any of the "$" label on it?

- [x] My DOT address: [Payout](https://canary.kodadot.xyz/dot/transfer/?target=1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)

### <samp>🤖 Generated by Copilot at 1962a9d</samp>

Improved the display and error handling of NFT images based on their mime types. Moved `useNftCardIcon` function to `ItemsGridImage` component for better performance.

### <samp>🤖 Generated by Copilot at 1962a9d</samp>

> _`useNftCardIcon`_
> _moved and simplified - faster_
> _autumn leaves falling_

### CODE REVIEW

**Comment by prury** [comment]
well done, thank you!

**Review by prury — APPROVED** [review]

**Review by Jarsen136 — APPROVED** [review]

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue [hunk 1]
```diff
@@ -26,14 +26,13 @@ defineProps<{
   isDarkMode?: boolean
 }>()
 
-// TODO: onError does infinite log in carousel type
```
Comment (roiLeo) [inline libs/ui/src/components/MediaItem/type/ImageMedia.vue:29]: hehehe beware of this one

**Review by roiLeo — APPROVED** [review]

File: libs/ui/src/components/MediaItem/type/ImageMedia.vue (hunk 1 above)
Comment (preschian) [reply libs/ui/src/components/MediaItem/type/ImageMedia.vue:29]: should be fixed with `.once`

Comment (roiLeo) [reply libs/ui/src/components/MediaItem/type/ImageMedia.vue:29]: Noice!

**Comment by preschian** [comment]
@prury I fixed another case, I just noticed yangwao and viiki seem like they use big screens
https://github.com/kodadot/nft-gallery/pull/7641#issuecomment-1761695559
https://github.com/kodadot/nft-gallery/pull/7636#issuecomment-1764218645

How to reproduce: zoom out the browser until the card is so small. visit https://canary.kodadot.xyz/rmrk/explore/items?listed=true

| before | after |
|--------|--------|
| Screenshot 2023-10-16 192506 | Screenshot 2023-10-16 192615 | 

@Jarsen136 @roiLeo, just in case you want to review it again, the changes are relatively small 1ec34b1416029c1d700daa1c3d6d0427a24bee32

**Comment by prury** [comment]
> @prury I fixed another case, I just noticed yangwao and viiki seem like they use big screens [#7641 (comment)](https://github.com/kodadot/nft-gallery/pull/7641#issuecomment-1761695559) [#7636 (comment)](https://github.com/kodadot/nft-gallery/pull/7636#issuecomment-1764218645)
> 
> How to reproduce: zoom out the browser until the card is so small. visit https://canary.kodadot.xyz/rmrk/explore/items?listed=true
> before 	after
> Screenshot 2023-10-16 192506 	Screenshot 2023-10-16 192615
> 
> @Jarsen136 @roiLeo, just in case you want to review it again, the changes are relatively small [1ec34b1](https://github.com/kodadot/nft-gallery/commit/1ec34b1416029c1d700daa1c3d6d0427a24bee32)

dudes using kodadot on a television, ha
nice finding! tested here and it worked fine

**Comment by yangwao** [comment]
pay 50 usd

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $50 @ 3.79 USD/DOT ~ 13.193 $DOT
🧗 [1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p](https://polkadot.subscan.io/account/1xjvRADwdJcnmUCLWerEHR4iGCT5EBTm4D4fzLLg4LcAC9p)
🔗 [0xe3601edeaf2476d28735210b4c1f76434b1389140ce986595fabe10b6a9fe72a](https://polkadot.subscan.io/extrinsic/0xe3601edeaf2476d28735210b4c1f76434b1389140ce986595fabe10b6a9fe72a)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**